crates/astria-core/src/sequencerblock/v1alpha1/celestia.rs

```diff
@@ -520,6 +520,11 @@ impl SubmittedMetadata {
         &self.header
     }
 
+    /// Returns the rollup IDs.
+    pub fn rollup_ids(&self) -> impl Iterator<Item = &RollupId> {
```

Comment:
Should we introduce a type that wraps this iterator (so that we don't return an anonymous type)?

Reply:
I'm not sure I see a benefit to that.  Is it a stylistic thing?

I could always change the return type to `std::slice::Iter<RollupId>`, but that's a bit less canonical than `impl Iterator` I think - it couples the struct's interface a bit more tightly to its private internals.

Reply:
The idea is to avoid impl trait in return position for public APIs (to not return anonymous types that is). It seems this actually never made it into the official Rust API guidelines. Here is a discussion that never lead anywhere, but I agree with burntsushi's point: https://github.com/rust-lang/api-guidelines/discussions/60

Reply:
Hmm.  Can't say I disagree with the points in general, but otoh I think the arguments against returning impl Trait are particularly weak in this specific case.  It seems unlikely that a consumer of this crate would need to hold an iterator of rolllup IDs in a struct or make it an associated type of their own trait.

I think it's reasonably common to see `impl Iterator` as a return type, and the fact that the linked discussion has relatively few comments/upvotes points to this not being an issue in general.

Saying that, the only real objection I have to your original suggestion is purely that it seems unnecessary - it clutters the code and the rustdocs, i.e. it's a very weak objection given how small a change it would be.  So, I'll just drop in a new `RollupIdIter` :)

Reply:
Done in [63a605c0](https://github.com/astriaorg/astria/pull/1059/commits/63a605c0a4c6c7006f22af34a00bb4bbf9fffec7).